### PR TITLE
[NextJs][Regression] Custom Components cannot be added using Horizon

### PR DESCRIPTION
**Pull request by CobyPear**
Attempting to edit a page in Horizon with a component created with `jss scaffold` in a Nextjs app causes an error. This change should alleviate that error.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

### CODE REVIEW

File: samples/react/package.json [hunk 1]
```diff
@@ -108,7 +108,7 @@
     "build:client:rendering-host": "cross-env-shell PUBLIC_URL=$npm_package_config_tunnelUrl \"react-scripts build\"",
     "build:server": "cross-env-shell NODE_ENV=production \"webpack --config server/server.webpack.config.js\"",
     "bootstrap": "node scripts/bootstrap.js",
-    "graphql:update": "node -r babel-register ./scripts/update-graphql-fragment-data.js",
```
Comment (ambrauer) [inline samples/react/package.json:111]: Looks like these react sample changes accidentally snuck in here?

File: samples/nextjs/scripts/templates/component-src.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ type ${componentName}Props = StyleguideComponentProps & {
 const ${componentName} = (props: ${componentName}Props): JSX.Element => (
   <div>
     <p>${componentName} Component</p>
-    <Text field={props.fields.heading} />
+    <Text field={props.fields?.heading} />
```
Comment (ambrauer) [inline samples/nextjs/scripts/templates/component-src.ts:19]: We should use the `withDatasourceCheck` HOC here, which provides datasource resiliency for Horizon. See `ContentBlock.tsx` for example.

**Review by ambrauer — CHANGES_REQUESTED** [review]

File: samples/react/package.json (hunk 1 above)
Comment (CobyPear) [reply samples/react/package.json:111]: Yeah haha I was just trying to revert this 😳

File: samples/nextjs/scripts/templates/component-src.ts (hunk 2 above)
Comment (CobyPear) [reply samples/nextjs/scripts/templates/component-src.ts]: Yes makes sense, done.

**Review by ambrauer — APPROVED** [review]

**Pull request merged**